docs/source/x_ext.rst

```diff
@@ -746,9 +769,10 @@ The only rule related to valid and ready signals is that:
 Specifically note the following:
 
 * The valid signals are allowed to be retracted by a |processor| (e.g. in case that the related instruction is killed in the |processor|'s pipeline before the corresponding ready is signaled).
+* A new transaction can be started by a |processor| by changing the ``id`` signal and keeping the valid signal asserted (thereby possibly terminating a previous transaction before it completed).
 * The valid signals are not allowed to be retracted by a |coprocessor| (e.g. once ``mem_valid`` is asserted it must remain asserted until the handshake with ``mem_ready`` has been performed).
-* A new transaction can be started by changing the ``id`` signal and keeping the valid signal asserted (thereby possibly terminating a previous transaction before it completed).
-* The ready signal is allowed to be 1 when the corresponding valid signal is not asserted.
+* A new transaction can not be started by a |coprocessor| by changing the ``id`` signal and keeping the valid signal asserted.
```

Comment:
> A new transaction can not be started by a |coprocessor| by changing the ``id`` signal and keeping the valid signal asserted.

I am not quite sure what your intentions are with this line, since it happens quite a lot that the coprocessor starts a new memory (or result) transaction just by changing the id and keeping valid asserted.
For example, it happens when the core offloads and commits two consecutive memory instructions. Then a coprocessor (or at least my floating-point coprocessor) will not dessert the valid signal of the memory request channel until both memory transactions are accepted by the core.

Reply:
This part of the spec is being misinterpreted. I agree it is not clear and will reword it a little bit. The point is that a transaction initiated by a coprocessor cannot be terminated by that coprocessor by just changing the id (as is possible for example on the issue interface). A coprocessor always has to wait for the corresponding ready signal (and when that comes a next transaction can start in the following cycle by just keeping valid high and picking a new id).


I will reword it a little bit. The

Reply:
ok, thanks for the clarification.